presto-common/src/main/java/com/facebook/presto/common/type/RowIdType.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.common.type;
+
+import com.facebook.presto.common.block.Block;
+import com.facebook.presto.common.block.BlockBuilder;
+import com.facebook.presto.common.function.SqlFunctionProperties;
+import io.airlift.slice.Slice;
+import io.airlift.slice.XxHash64;
+
+import static com.facebook.presto.common.type.TypeSignature.parseTypeSignature;
+
+/**
+ * TODO this is wrong. need a binary format
+ * A row ID is a unique identifier for o row that is a binary string in
+ * this format:
+ * <p>
+ * row_number:row_group_id:data_version:partition_id:table_id:table_guid.substring(0, 4)
+ *
+ * <ul>
+ * <li>row_number: a 64 bit long, unique within the partition</li>
+ * <li>row_group_id: the file name</li>
+ * <li>data_version: version of the partition</li>
+ * <li>partition_id: a 64 bit long</li>
+ * <li>table_id: a 64 bit long</li>
+ * </ul>
+ * <p>
+ * The Presto blocks holding this data encode them as slices; that is, UTF-8 strings.
+ */
+public class RowIdType
+        extends AbstractVariableWidthType
+{
+    public static final RowIdType ROW_ID = new RowIdType();
+
+    private RowIdType()
+    {
+        super(parseTypeSignature(StandardTypes.ROW_ID), Slice.class);
+    }
+
+    @Override
+    public Object getObjectValue(SqlFunctionProperties sqlFunctionProperties, Block block, int position)
+    {
+        if (position < 0) {
+            throw new IndexOutOfBoundsException("Negative block position " + position);
+        }
+        if (block.getPositionCount() == 0 || block.isNull(position)) {
+            return null;
+        }
+        return block.getSlice(position, 0, block.getSliceLength(position)).toStringUtf8();
```

Comment:
I meant why did you choose to treat this type as a string instead of a binary. I'd suggest to have the same implementation as VarbinaryType, unfortunately we cannot just extend it so copy-paste it is. 
In this case of getObjectValue method it would return a SqlVarbinary.

Reply:
Treating it as a string was probably just an early misunderstanding on my part. SQLVarBinary is fine. Done